prime-router/src/main/kotlin/fhirengine/engine/FHIRConverter.kt

```diff
@@ -137,68 +140,92 @@ class FHIRConverter(
                     )
 
                     maybeParallelize(
-                        fhirBundles.size,
-                        Streams.mapWithIndex(fhirBundles.stream()) { bundle, index ->
+                        processedItems.size,
+                        Streams.mapWithIndex(processedItems.stream()) { bundle, index ->
                             Pair(bundle, index)
                         },
                         "Applying sender transforms and routing"
-                    ).map { (bundle, bundleIndex) ->
+                    ).map { (processedItem, itemIndex) ->
                         // conduct FHIR Transform
-                        transformer?.process(bundle)
-
-                        // make a 'report'
-                        val report = Report(
-                            MimeFormat.FHIR,
-                            emptyList(),
-                            parentItemLineageData = listOf(
-                                Report.ParentItemLineageData(queueMessage.reportId, bundleIndex.toInt() + 1)
-                            ),
-                            metadata = this.metadata,
-                            topic = queueMessage.topic,
-                            nextAction = TaskAction.destination_filter
-                        )
+                        if (processedItem.bundle == null) {
+                            val report = Report(
+                                MimeFormat.FHIR,
+                                emptyList(),
+                                parentItemLineageData = listOf(
+                                    Report.ParentItemLineageData(queueMessage.reportId, itemIndex.toInt() + 1)
+                                ),
+                                metadata = this.metadata,
+                                topic = queueMessage.topic,
+                                nextAction = TaskAction.none
+                            )
+                            val noneEvent = ProcessEvent(
+                                Event.EventAction.NONE,
+                                report.id,
+                                Options.None,
+                                emptyMap(),
+                                emptyList()
+                            )
+                            actionHistory.trackCreatedReport(noneEvent, report)
+                            null
+                        } else {
+                            // We know from the null check above that this cannot be null
+                            val bundle = processedItem.bundle!!
+                            transformer?.process(bundle)
+
+                            // make a 'report'
+                            val report = Report(
+                                MimeFormat.FHIR,
+                                emptyList(),
+                                parentItemLineageData = listOf(
+                                    Report.ParentItemLineageData(queueMessage.reportId, itemIndex.toInt() + 1)
```

Comment:
Why is the `parentReportIndex` the `itemIndex + 1`?

Reply:
Are index tracking always starts at 1 rather than zero